includes/REST_API/Stories_Controller.php

```diff
@@ -165,18 +165,47 @@ public function get_item_schema() {
 	 */
 	public function filter_posts_orderby( $orderby, $query ) {
 		global $wpdb;
-		if ( Story_Post_Type::POST_TYPE_SLUG !== $query->get( 'post_type' ) ) {
+		if ( ! $this->should_filter_query( $query ) ) {
 			return $orderby;
 		}
-		if ( 'story_author' !== $query->get( 'orderby' ) ) {
-			return $orderby;
+		$order = $query->get( 'order' );
+
+		return "$wpdb->users.display_name $order, $wpdb->posts.post_modified $order"; // phpcs:ignore WordPressVIPMinimum.Variables.RestrictedVariables.user_meta__wpdb__users
+	}
+
+	/**
+	 * Filters the join query to filter first all the current user's posts and then the rest.
+	 *
+	 * @param string    $join Join clause.
+	 * @param \WP_Query $query WP_Query object.
+	 * @return string Join clause.
+	 */
+	public function filter_posts_join( $join, $query ) {
+		global $wpdb;
+		if ( ! $this->should_filter_query( $query ) ) {
+			return $join;
 		}
 
-		$current_user = get_current_user_id();
-		if ( ! $current_user ) {
-			return $orderby;
+		$join .= " LEFT JOIN $wpdb->users ON ($wpdb->posts.post_author = $wpdb->users.ID) "; // phpcs:ignore WordPressVIPMinimum.Variables.RestrictedVariables.user_meta__wpdb__users
+
+		return $join;
+	}
+
+	/**
+	 * Check if query should be filtered.
+	 *
+	 * @param \WP_Query $query WP_Query object.
+	 * @return bool
+	 */
+	public function should_filter_query( $query ) {
```

Comment:
Pasting my feedback from Slack:

When using the more convenient `posts_clauses` filter, there's no need for this method because those checks will only be needed once.